examples-gallery/plot_particle_in_tube.py

```diff
@@ -0,0 +1,533 @@
+state_symbols = (x, y, z, vx, vy, vz, cut_param, dist, cutdt, gate)
+specified_symbols = (Fx, Fy, Fz, gate_h)
+
+t0, duration = 0, 5.0
+num_nodes = 101
+interval_value = duration/(num_nodes - 1)
+tm = interval_value*50
+
+# %%
+# Provide some values for the constants.
+radius = 1.0
+max_z = 12.0
+start_param = 0.001
+par_map = {
+    c: 0.5*0.1*1.2,
+    g: 9.81,
+    m: 2.0,
+    a1: 5.0,
+    a2: 5.0,
+    a3: 5.0,
+}
+
+# %%
+# Specify the objective function and form the gradient.
+obj_func = sm.Integral(Fx**2 + Fy**2 + Fz**2, t)
+
+sm.pprint(obj_func)
+obj, obj_grad = create_objective_function(
+    obj_func,
+    state_symbols,
+    specified_symbols,
+    tuple(),
+    num_nodes,
+    interval_value,
+    time_symbol=t,
+)
+
+# %%
+# Specify the symbolic instance constraints.
+eval_curve = sm.lambdify((r, a1, a2, a3), curve, cse=True)
+
+instance_constraints = (
+    # start level
+    x.func(0.0) - eval_curve(start_param, par_map[a1], par_map[a2],
+                             par_map[a3])[0],
+    y.func(0.0) - eval_curve(start_param, par_map[a1], par_map[a2],
+                             par_map[a3])[1],
+    z.func(0.0) - eval_curve(start_param, par_map[a1], par_map[a2],
+                             par_map[a3])[2],
+    cut_param.func(0.0) - start_param,
+    dist.func(0.0),
+    # intermediate level
+    gate.func(tm) - radius/4.0,
+    # end level
+    x.func(duration) - eval_curve(max_z/par_map[a3], par_map[a1], par_map[a2],
+                                  par_map[a3])[0],
+    y.func(duration) - eval_curve(max_z/par_map[a3], par_map[a1], par_map[a2],
+                                  par_map[a3])[1],
+    z.func(duration) - max_z,
+    # stationary at start and finish
+    vx.func(0.0),
+    vy.func(0.0),
+    vz.func(0.0),
+    vx.func(duration),
+    vy.func(duration),
+    vz.func(duration),
+)
+print('len(instance_constraints) =', len(instance_constraints))
+
+# %%
+# Add some physical limits to the force, other bounds as needed to realize
+# the inequalities.
+grenze = 100.0
+bounds = {
+    Fx: (-grenze, grenze),
+    Fy: (-grenze, grenze),
+    Fz: (-grenze, grenze),
+    dist: (0.0, radius),
+    cut_param: (0.0, np.inf),
+    z: (0.0, max_z),
+    cutdt: (0.0, 5.0),
+    gate_h: (1.0, 10.0),
+}
+
+# %%
+# Create the optimization problem and set any options.
+prob = Problem(
+    obj,
+    obj_grad,
+    eom,
+    state_symbols,
+    num_nodes, interval_value,
+    known_parameter_map=par_map,
+    instance_constraints=instance_constraints,
+    bounds=bounds,
+    time_symbol=t,
+)
+
+prob.add_option('max_iter', 3000)
+
+# %%
+# Give a guess of a plausible route with constant thrust. Here the
+# solution of a previous run is used as initial guess.
+initial_guess = np.zeros(prob.num_free)
+x_guess, y_guess, z_guess = eval_curve(np.linspace(0.0, max_z/par_map[a3],
+                                                   num=num_nodes), par_map[a1],
+                                       par_map[a2], par_map[a3])
```

Comment:
The difference indentation matches with the leading parentheses for the function the arguments are related to.